model/flow/transaction_test.go

```diff
@@ -58,3 +69,554 @@ func TestTransaction_Status(t *testing.T) {
 		assert.Equal(t, status.String(), value)
 	}
 }
+
+// TestTransactionAuthenticationSchemes tests transaction signature verifications
+// with a focus on authentication schemes.
+func TestTransactionAuthenticationSchemes(t *testing.T) {
```

Comment:
where these a direct move? i.e. don't have to re-validate any of this logic i assume?